### PR TITLE
Standardized JSON error responses

### PR DESCRIPTION
**Pull request by jlaswell**
Addition of a jsonutil package to help manage error responses for api responses. Handlers in the api section have been refactored to use this new package. A field tag was added to the ID field of the base database.Model struct to force json marshalling to return a string.

### CODE REVIEW

File: app/handlers/jsonutil/json.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+// Status acts as a getter to provide the high-level status.
+func (jre *JSONResponseError) Status() int {
+	return jre.status
+}
+
+// AddError adds a new ResponseError, while accepting an optional status code
+// specific to the new ResponseError.
+func (jre *JSONResponseError) AddError(details string, status ...int) *JSONResponseError {
```
Comment (kyleferguson) [inline app/handlers/jsonutil/json.go:67]: What's the point of accepting a variadic status code when we're just going to use the first one if multiple are passed?

Comment (kyleferguson) [reply app/handlers/jsonutil/json.go:67]: Never mind I see it's for an optional param.

**Review by kyleferguson — APPROVED** [review]
I think there's probably a better way to handle utils like this but it looks good to me as it is, we can look at refactoring later. Nice work!

**Comment by jlaswell** [comment]
I'm 100% for finding a better way to organize all of this. 😆 I feel like I'm just making this more and more cluttered, so an organizational refactor would be great before too long.

**Pull request merged**